mycroft/skills/skill_manager.py

```diff
@@ -257,20 +247,23 @@ def install_or_update(skill):
                         nonlocal default_skill_errored
                         default_skill_errored = True
                     raise
-                installed_skills.add(skill.name)
+            installed_skills.add(skill.name)
 
         try:
-            self.msm.apply(install_or_update, self.msm.list())
+            with self.msm.lock, self.thread_lock:
+                self.msm.skills_data = self.msm.load_skills_data()
```

Comment:
Is this and the still needed now that it performs `sync_skills_data` in the constructor? Additionally, would it cause a deadlock? (I haven't checked to see if nested locks will deadlock with filelock)